pkg/noderesourcetopology/least_allocatable.go

```diff
@@ -0,0 +1,38 @@
+package noderesourcetopology
+
+import (
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+)
+
+func leastAllocatableScoreStrategy(requested, allocatable v1.ResourceList, resourceToWeightMap resourceToWeightMap) int64 {
+	var numaNodeScore int64 = 0
+	var weightSum int64 = 0
+
+	for resourceName := range requested {
+		// We don't care what kind of resources are being requested, we just iterate all of them.
+		// If NUMA zone doesn't have the requested resource, the score for that resource will be 0.
+		resourceScore := leastAllocatableScore(requested[resourceName], allocatable[resourceName])
+		weight := resourceToWeightMap.weight(resourceName)
+		numaNodeScore += resourceScore * weight
+		weightSum += weight
+	}
+
+	return numaNodeScore / weightSum
+}
+
+// The used capacity is calculated on a scale of 0-MaxNodeScore (MaxNodeScore is
+// constant with value set to 100).
+// 0 being the lowest priority and 100 being the highest.
+// The less allocatable resources the node has, the higher the score is.
+func leastAllocatableScore(requested, numaCapacity resource.Quantity) int64 {
+	if numaCapacity.CmpInt64(0) == 0 {
+		return 0
+	}
+	if requested.Cmp(numaCapacity) > 0 {
+		return 0
+	}
+
+	return (requested.Value() * framework.MaxNodeScore) / numaCapacity.Value()
```

Comment:
Apologize that I should have spotted it earlier.

Should we use the term "Allocated" (aligned with the upstream plugins) or "Allocatable"? My 2 cents is "Allocated". WDYT? If you agree, the logic here should be `capacity - requested`, and vice versa for "mostAllocated". Also the string liternals in types.go should be updated.

Reply:
Actually, I was trying to be aligned with the naming convention of this repo's plugin:
For example here https://github.com/kubernetes-sigs/scheduler-plugins/tree/master/pkg/noderesources we're also using the terms "Allocatable", "leastAllocatable" etc. so I thought it would be more accurate to keep the same convention. 

But again, if you think it would be better to follow the upstream plugins convention, I'll do so.

Reply:
Oh, this repo's **allocatable** plugin was an supplement to the upstream plugins, please see the motivations in its KEP. And this PR tries to introduce regular plugins for topology aware scenarios, so I'd prefer to rename to "xyzallocated". Thanks.